swingbits/src/main/java/org/oxbow/swingbits/list/CheckList.java

```diff
@@ -91,6 +96,18 @@ private CheckList(final JList list) {
         setupKeyboardActions(list);
 
     }
+    
+    
+    private CheckList(JList list, CheckListRenderer customRenderer) {
+        if (list == null) throw new NullPointerException();
+        this.list = list;
+        this.list.getSelectionModel().setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+        
+        if ( !isEditorAttached() ) list.addMouseListener(checkBoxEditor);
+        this.list.setCellRenderer(customRenderer);

```

Comment:
It is better to use `setCellRenderer` method, since you already have it, instead of duplicating the code.